aptos-move/aptos-vm/src/verifier/transaction_arg_validation.rs

```diff
@@ -184,6 +185,127 @@ pub fn validate_combine_signer_and_txn_args(
     Ok(combined_args)
 }
 
+pub(crate) fn validate_combine_signer_and_txn_args_for_batched_calls(
```

Comment:
Would love to see how we can integrate this more "natively" without code duplication